transactor/src/main/scala/io/mediachain/transactor/StateMachine.scala

```diff
@@ -60,10 +63,10 @@ object StateMachine {
     private def insertRecord(cmd: JournalInsert): Xor[JournalError, CanonicalEntry] = {
       val rec = cmd.record
       val ref = datastore.put(rec)
-      index.get(ref) match {
+      state.index.get(ref) match {
```

Comment:
are you talking about a local index variable?
Sure, It could, but I kind of prefer to have it obvious where the state is modified.